test/release/examples/primers/arrays.chpl

```diff
@@ -339,20 +351,35 @@ VarDom = {1..n};
 writeln("VarArr should now be reset: ", VarArr, "\n");
 
 //
-// Note that querying an array's domain via the ``.domain`` method or
-// the function argument query syntax does not result in a domain
-// expression that can be reassigned.  In particular, we cannot do:
+// This trick can be used to create records or classes with array fields which
+// may have different sizes in each instance of the type:
```

Comment:
Calling this a "trick" (declaring a named domain) seems weird to me.  "This approach"?  "This technique"?

There also feels like there's a bit of a leap between "domains can be declared and resized" and this approach, which could be softened by transitioning with a phrase like "By using domain fields within a record, the user can also create objects whose storage can dynamically vary." rather than the previous approaches.

That said, note that one can do the example below without a domain field simply by doing `const size: int;` rather than `var FieldDom = {1..0};` (and `FieldDom` could also be `const`).  Cases where declaring a `FieldDom` (and as `var` are important are ones where the array needs to be resized after its initialization point.

So what would I do about this...  I might either transition to this example by declaring the `const size: int; var FieldArr: [1..size] int;` case first then point out that it can't be resized after initialization time;  or I might just beef up this example by having it size or resize the array after the initialization code.  (at which point it could then comment that if the array's size only needs to be set at initialization time, the user could get the effect without an explicit domain field by using an `int` size field).

Reply:
> Calling this a "trick" (declaring a named domain) seems weird to me. "This approach"? "This technique"?

Agreed. "Trick" is used earlier in the primer, I wanted to stick to that language. But I'll adjust.

> or I might just beef up this example by having it size or resize the array after the initialization code. (at which point it could then comment that if the array's size only needs to be set at initialization time, the user could get the effect without an explicit domain field by using an int size field).

I like resizing as you suggest. I suspect an `int`-field based approach is more about a record trick rather than an array trick, so I am hesitant to refer to something like that.

Reply:
Oh, I see now.  I was just reading the diff (quickly) and didn't realize that "This trick" referred to something previously introduced rather than the example that followed the colon.  (i.e., "The following trick" is how I read it).  Now that I've got the context, I'm more OK with it.  But it also feels like a different trick.  The antecedent trick is "What if I want to reallocate an array without paying the cost of preserving its elements (which is very expensive in practice?" whereas this one is "How can I create objects with varying array sizes?"  (or variable array sizes?) — i.e., there's no re-assignment to {1..0} used or needed, as in the previous trick.